api/src/test/java/org/apache/commons/rdf/api/AbstractGraphTest.java

```diff
@@ -228,6 +231,128 @@ public void getTriplesQuery() throws Exception {
         assertEquals(3, graph.getTriples(null, member, null).count());
     }
 
+    @Test
+    public void addBlankNodesFromMultipleGraphs() {
+
+        try {
+            Graph g1 = createGraph1();
+            Graph g2 = createGraph2();
+            Graph g3 = factory.createGraph();
+
+            copyTriples(g1, g3);
+            copyTriples(g2, g3);
+
+            IRI name = factory.createIRI("http://xmlns.com/foaf/0.1/name");
+
+            final Map<String, BlankNodeOrIRI> whoIsWho = new HashMap<>();
+            System.out.println(g3.getTriples(null, name, null).count());
+            // NOTE: sequential as our HashMap is not thread-safe
```

Comment:
Once the HashMap above is changed to a ConcurrentHashMap, there should be no reason why parallel shouldn't work given we aren't modifying the Graph concurrently. An intermittent failure like that is something we need to look into further as it could be a bug in the actual code, assuming it still occurs with ConcurrentHashMap (as it is clear why it would happen with HashMap.put but not with ConcurrentHashMap.put).

There shouldn't be any reason why accessing a Graph in parallel from multiple threads through a single Stream shouldn't work, as long as the Graph isn't being modified, so we need to look into why that might be occuring in copyTriple as well.